libraries/c_sdk/standard/mqtt/src/iot_mqtt_subscription.c

```diff
@@ -536,54 +561,76 @@ void _IotMqtt_RemoveSubscriptionByTopicFilter( _mqttConnection_t * pMqttConnecti
-        topicMatchParams.exactMatchOnly = true;
+        /* Find and remove each topic filter from the list. */
+        for( i = 0; i < subscriptionCount; i++ )
+        {
+            topicMatchParams.pTopicName = pSubscriptionList[ i ].pTopicFilter;
+            topicMatchParams.topicNameLength = pSubscriptionList[ i ].topicFilterLength;
+            topicMatchParams.exactMatchOnly = true;
 
-        pSubscriptionLink = IotListDouble_FindFirstMatch( &( pMqttConnection->subscriptionList ),
-                                                          NULL,
-                                                          _topicMatch,
-                                                          &topicMatchParams );
+            matchedIndex = IotMqtt_FindFirstMatch( &( connToContext[ contextIndex ].subscriptionArray ), 0,
+                                                   &topicMatchParams );
 
-        if( pSubscriptionLink != NULL )
-        {
-            pSubscription = IotLink_Container( _mqttSubscription_t, pSubscriptionLink, link );
+            if( matchedIndex != -1 )
+            {
+                pSubscription = &( connToContext[ contextIndex ].subscriptionArray[ matchedIndex ] );
 
-            /* Reference count must not be negative. */
-            IotMqtt_Assert( pSubscription->references >= 0 );
+                /* Reference count must not be negative. */
+                IotMqtt_Assert( pSubscription->references >= 0 );
 
-            /* Remove subscription from list. */
-            IotListDouble_Remove( pSubscriptionLink );
+                /* Remove subscription from list. */
+                if( IotMqtt_RemoveSubscription( ( connToContext[ contextIndex ].subscriptionArray ), matchedIndex ) == false )
+                {
+                    subscriptionStatus = false;
+                }
 
-            /* Check the reference count. This subscription cannot be removed if
-             * there are subscription callbacks using it. */
-            if( pSubscription->references > 0 )
-            {
-                /* Set the unsubscribed flag. The last active subscription callback
-                 * will remove and clean up this subscription. */
-                pSubscription->unsubscribed = true;
+                IotMqtt_Assert( subscriptionStatus == true );
+
+                /* Check the reference count. This subscription cannot be removed if
+                 * there are subscription callbacks using it. */
+                if( pSubscription->references > 0 )
+                {
+                    /* Set the unsubscribed flag. The last active subscription callback
+                     * will remove and clean up this subscription. */
+                    pSubscription->unsubscribed = true;
+                }
+                else
+                {
+                    /* Free a subscription by setting topicFilterLength to 0 as it indiactes that subscription is unsubscribed. */
```

Comment:
please check all the spellings in this PR. indiactes -> indicates

Reply:
Addressed